front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -0,0 +1,162 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { Authenticator, getSession } from "@app/lib/auth";
+import { getOrCreateSystemApiKey } from "@app/lib/auth";
+import { ConnectorsAPI } from "@app/lib/connectors_api";
+import { DustAPI } from "@app/lib/dust_api";
+import { DataSource, Key, Provider } from "@app/lib/models";
+import { credentialsFromProviders } from "@app/lib/providers";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "POST":
+      if (!auth.isBuilder()) {
+        return apiError(req, res, {
+          status_code: 401,
+          api_error: {
+            type: "app_auth_error",
+            message:
+              "Only the users that are `builders` for the current workspace can create a data source.",
+          },
+        });
+      }
+
+      const dataSourceName = "Slack (managed)";
```

Comment:
cc @fontanierh I think you designed against `managed-{slack|...}` right? let's use that here as well.